src/Adapters/Storage/Mongo/MongoStorageAdapter.js

```diff
@@ -0,0 +1,48 @@
+
+let mongodb = require('mongodb');
+let MongoClient = mongodb.MongoClient;
+
+export class MongoStorageAdapter {
+  // Private
+  _uri: string;
+  // Public
+  connectionPromise;
+  database;
+
+  constructor(uri: string) {
+    this._uri = uri;
+  }
+
+  connect() {
+    if (this.connectionPromise) {
+      return this.connectionPromise;
+    }
+
+    this.connectionPromise = MongoClient.connect(this._uri).then(database => {
+      this.database = database;
+    });
+    return this.connectionPromise;
+  }
+
+  collection(name: string) {
+    return this.connect().then(() => {
+      return this.database.collection(name);
+    });
+  }
+
+  collectionsContaining(match: string) {
```

Comment:
Can you add a comment that this is for clearing out collections during tests? I was kinda confused about why this function exists.